GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -3309,6 +3356,15 @@ private void UpdateButtonStates()
                 : TranslatedStrings.ButtonPush;
         }
 
+        private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
+        {
+            ThreadHelper.FileAndForget(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+                    RescanChanges();
```

Comment:
Other explicit calls of `RescanChanges()` might not be needed any longer.

Reply:
Maybe, but it's difficult for me to figure out which is necessary, which isn't, and which is necessary some times. Such a clean up can be handled by a separate PR anyway.

Reply:
Unnecessary refreshes in FormCommit result in a bad UX - mainly for un/staging. This is not a just nice-to-have cleanup but mandatory.
If we are lucky, it suffices to check for `_skipUpdate` and `_bypassActivatedEventHandler`.

Reply:
If there's an easy way to do it, I'm listening. What I'm considering doing, is replacing all the `_bypassActivatedEventHandler`s and `_shouldRescanChanges`s (`_skipUpdate` seems unrelated) with a simple dirty queue/flush system, which basically does a rescan after all current actions are done. This would guarantee no double rescans.

What confuses me, though, is the use of `InitializedStaged` which is basically a competing `RescanChanges`. I'm not sure what each does. There's also a bunch of places where `RescanChanges` is suppressed, and then `Initialize` is called directly.

This is complex and I can't make heads and tails of it. Not the least bit because, despite your statement about bad UX, there's no visible way to confirm that a rescan has happened twice.

Also, the code there is such a mess, I can't tell if double rescan isn't happening already.

Reply:
FormCommit has a long history...
I did not want to trigger a rewrite of it.
The following might suffice:
```
if (!_skipUpdate && !_bypassActivatedEventHandler)
{
    FileAndForget({... RescanChanges(); ...});
}
```

Reply:
Accepted